frontend/src/components/login-form/login-form.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import Button from '../button/button';
+import Form from '../input-form/input-form';
+import './login-form.css';
+import { ENDPOINTS } from '../../api/config';
+
+const LoginForm = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [role, setRole] = useState('Startup'); // Default role
+  const [error, setError] = useState(null); // Error handling
+  const navigate = useNavigate();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setError(null);
+
+    // Determine the role value (1 for Startup, 2 for Investor)
+    const roleValue = role === 'Startup' ? 1 : 2;
+    // Prepare the request payload
+    const payload = {
+      role: roleValue,
+      email,
+      password,
+    };
+
+    try {
+      const response = await fetch(`${ENDPOINTS.LOGIN}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(payload),
+      });
+
+      if (!response.ok) {
+        throw new Error('Login failed. Please check your credentials.');
+      }
+
+      const data = await response.json();
+
+      // Save JWT tokens in localStorage
+      localStorage.setItem('accessToken', data.access);
+      localStorage.setItem('refreshToken', data.refresh);
+
+      // Redirect to the profile page
+      navigate('/my_profile/');
+    } catch (error) {
+      setError(error.message || 'Something went wrong. Please try again.');
+    }
+  };
```

Comment:
Provide feedback to the user while the login request is being processed.
```
const [isLoading, setIsLoading] = useState(false);

const handleSubmit = async (e) => {
  e.preventDefault();
  setError(null);
  setIsLoading(true);

  try {
    // Same logic as before...
    setIsLoading(false);
    navigate('/my_profile/');
  } catch (error) {
    setIsLoading(false);
    setError(error.message || 'Something went wrong. Please try again.');
  }
};
```

in JSX

```{isLoading ? <p>Loading...</p> : <Button text="Sign In" variant="primary" />}```